app/services/proofing/resolution/plugins/aamva_plugin.rb

```diff
@@ -36,15 +36,36 @@ def call(
               applicant_pii
             end
 
-          timer.time('state_id') do
+          first_result = timer.time('state_id') do
             proofer.proof(applicant_pii_with_state_id_address)
-          end.tap do |result|
-            if result.exception.blank?
-              Db::SpCost::AddSpCost.call(
-                current_sp,
-                :aamva,
-                transaction_id: result.transaction_id,
-              )
+          end
+
+          if first_result.exception.blank?
+            Db::SpCost::AddSpCost.call(
+              current_sp,
+              :aamva,
+              transaction_id: first_result.transaction_id,
+            )
+          end
+
+          if !first_result.success? && first_result.errors[:first_name] == 'UNVERIFIED'
+            names = applicant_pii_with_state_id_address[:first_name].split(' ')
+            if names.size == 2
```

Comment:
just want to drop this classic: [Falsehoods programmers believe about names](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)

Having to normalize a combined field like this will always have errors but I think we can reduce a few of them by allowing for people with multiple first names (without a hyphen).

If say, "John Paul Smith Jones" was our person, and Socure returns them as first name "John Paul Smith" then this would give us "John Paul" and "Smith"

I might suggest:

```suggestion
            names = applicant_pii_with_state_id_address[:first_name].split(' ')
            if names.size > 1
              *first_names, middle = names
              first_name = first_names.join(' ')
```

Reply:
And this assumes people only have one middle name! Which I'm sure is also false

Reply:
Yes, and if someone has one first name but multiple middle names we have the same problem. Not to mention we don't know how the user's MVA is storing their name. Happily this PR was closed since the observed issue was fixed on Socure's end.